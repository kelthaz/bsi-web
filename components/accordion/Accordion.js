/* eslint-disable jsx-a11y/no-static-element-interactions */

import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styles from './accordion.module.scss';

const Accordion = (props) => {
  const { children, expanded, title } = props;

  const [expand, setExpand] = useState(expanded);

  const handleAccordeon = () => {
    setExpand(!expand);
  };

  return (
    <>
      <button type="button" className={`${styles.accordion} ${expand ? styles.active : ''}`} onClick={handleAccordeon}>
        {title}
      </button>
      <div className={`${styles.panel} ${expand ? styles['panel-active'] : ''}`}>{children}</div>
    </>
  );
};

Accordion.propTypes = {
  title: PropTypes.any.isRequired,
  expanded: PropTypes.bool.isRequired,
  children: PropTypes.any.isRequired,
};

export default Accordion;
