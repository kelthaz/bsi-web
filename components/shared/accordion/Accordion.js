import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styles from './accordion.module.scss';

const Accordion = (props) => {
  const { children, expanded, title, color, icon } = props;

  const styleAccordion = styles[`accordion-${color}`];
  const styleIcon = styles[`${icon}-icon-${color}`];
  const styleActiveAccordion = styles[`active-${icon}`];
  const [expand, setExpand] = useState(expanded);

  const handleAccordeon = () => {
    setExpand(!expand);
  };

  return (
    <div className={color === 'blue' ? styles.border : ''}>
      <button
        type="button"
        className={`${styleAccordion} ${styleIcon} ${expand ? styleActiveAccordion : ''}`}
        onClick={handleAccordeon}
      >
        {title}
      </button>
      <div className={`${styles.panel} ${expand ? styles['panel-active'] : ''}`}>{children}</div>
    </div>
  );
};

Accordion.propTypes = {
  title: PropTypes.any.isRequired,
  expanded: PropTypes.bool.isRequired,
  children: PropTypes.any.isRequired,
  color: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
};

export default Accordion;
