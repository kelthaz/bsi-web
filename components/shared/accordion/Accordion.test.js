import React from 'react';
import { shallow } from 'enzyme';
import Accordion from './Accordion';
// import '@testing-library/jest-dom';

describe('Pruebas en el componente Accordion', () => {
  test('Debe crear la prueba instantanea', () => {
    // arrange
    const props = {
      title: 'Acerca de BanCoppel',
      expanded: false,
      color: 'white',
      icon: 'arrow',
    };
    // act
    const wrap = shallow(
      <Accordion {...props}>
        <h1>Welcome to Next.js!</h1>
      </Accordion>
    );
    // assert
    expect(wrap).toMatchSnapshot();
  });

  test('Debe mostrar Welcome to Next.js! dentro del componente', () => {
    const wrap = shallow(
      <Accordion title="Acerca de BanCoppel" expanded={false} color="white" icon="arrow">
        <h1>Welcome to Next.js!</h1>
      </Accordion>
    );
    expect(wrap.find('h1').text()).toEqual('Welcome to Next.js!');
  });
});
