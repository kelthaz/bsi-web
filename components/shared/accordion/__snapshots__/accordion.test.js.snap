// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Accordion Debe crear la prueba instantanea 1`] = `
<div
  className=""
>
  <button
    className="accordion-white arrow-icon-white "
    onClick={[Function]}
    type="button"
  >
    Acerca de BanCoppel
  </button>
  <div
    className="panel "
  >
    <h1>
      Welcome to Next.js!
    </h1>
  </div>
</div>
`;
